synapse/federation/sender/__init__.py

```diff
@@ -447,7 +447,7 @@ async def handle_event(event: EventBase) -> None:
                             )
                         )
 
-                        if len(partial_state_destinations) > 0:
+                        if partial_state_destinations is not None:
```

Comment:
I initially thought there was a bug here, where we would block on the full state of the room (below) when the server we joined off returned an empty list of servers in the room. But while writing this PR, I discovered that we validate that the list is truthy and so the bug can't happen.